### ISSUE TITLE
[Bazel 7.0.0] py_test will not generate test.xml under bazel-testlogs if test is cached remotely. 

### ISSUE THREAD

**Issue by nlou9** [bazelbuild/bazel#20913]
### Description of the bug:

With Bazel `7.0.0`, py_test will not generate `test.xml` under `bazel-testlogs` folder if test is cached. It generates the files below but not test.xml. 
```
dr-xr-xr-x  2 nlou  wheel    64 Jan 16 15:41 test.outputs
drwxr-xr-x  2 nlou  wheel    64 Jan 16 15:41 test.outputs_manifest
drwxr-xr-x  2 nlou  wheel    64 Jan 16 15:41 test.raw_splitlogs
-r-xr-xr-x  1 nlou  wheel  6962 Jan 16 15:41 test.log
-r-xr-xr-x  1 nlou  wheel   180 Jan 16 15:41 test.cache_status
```

It worked ok with bazel 6.4.0

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

run bazel test with a py_test target. 
test options :
```
build --enable_bzlmod=false
build --remote_download_minimal
build --nobuild_runfile_links
test --experimental_fetch_all_coverage_outputs
test --experimental_split_coverage_postprocessing
```

### Which operating system are you running Bazel on?

macos, linux

### What is the output of `bazel info release`?

7.0.0

**Comment by coeuvre** [comment]
I think this is WAI since you built with `--remote_download_minimal`. Can you try `--remote_download_toplevel` instead?

**Comment by mkosiba** [comment]
we noticed this as well and are also interested in a way to ensure `test.xml` is downloaded with `--remote_download_minimal`.

`--remote_download_toplevel` is not a great workaround in our case as it causes commands like `bazel test //...` to download a lot of additional files leading to increased CI times and cache load.

**Comment by tjgq** [comment]
Have you tried the `--remote_download_regex` workaround suggested above?

Minimal should be, well, minimal. I don't think we can't change its definition to fit your use case without breaking other people's use cases.

**Comment by tjgq** [comment]
Yes, the `--remote_download_*` options are interpreted as a union, not an intersection.

I acknowledge that `--remote_download_regex` isn't a perfect solution, but it's also not reasonable to add flags for all possible combinations of outputs one might want to download. In the longer term, we expect a [remote output service](https://github.com/bazelbuild/bazel/discussions/20933) to be a viable alternative for lazily downloading arbitrary outputs.

**Comment by illicitonion** [comment]
While I acknowledge we can't add all possible flags, it could be useful in the short/medium term to add a `--remote_download_mnemonics` flag?

I find that in practice mnemonic is actually the thing I want to filter on - generated code, java compile or java source jar actions, TestRunner outputs - they're all mnemonic-based filtering, and the remote execution stack already has the mnemonic handy when running actions...

**Comment by joeljeske** [comment]
We also noticed this change in behavior going from Bazel 6.4.0 to 7.1.1. I am fine with adding the remote_download_regex flag

I am curious, are there any other file types that Bazel no longer downloads some files under minimal mode?

**Comment by cbandera** [comment]
Same here. I can back the statements that came before.
Using the regex works and is a feasible workaround.
But I also like the idea of filtering by `mnemonic`.
